### PR TITLE
TypeScript syntax highlighting

### PR DESCRIPTION
**Pull request by jhpratt**
#1449 

I probably missed something, but it just adds in a few keywords from JS.

### CODE REVIEW

**Comment by elextr** [comment]
This could probably go on the wiki with the other filetype configurations [here](https://wiki.geany.org/filetype?do=showtag&tag=filetype)

**Comment by jhpratt** [comment]
Why not just make it built-in? The vast majority of users will not be searching for syntax highlighting - I didn't even know it was possible to make a custom filetype until I ran into this.

**Comment by elextr** [comment]
No typescriptists willing to test?

**Comment by aplumb-neurala** [comment]
I tried out the changes here and they work fine for me.

**Comment by marcooliva** [comment]
Please, how can i install this TypeScript syntax highlighting in Geany?

File: data/filedefs/filetypes.TypeScript.conf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# based on JavaScript file
```
Comment (ntrel) [inline data/filedefs/filetypes.TypeScript.conf]: This file needs to be listed in `data/Makefile.am`.

**Comment by ntrel** [comment]
@jhpratt Plus resolve the conflict please if possible. Thanks for submitting this BTW.

@marcooliva Download the `filetypes.TypeScript.conf` file and put it in either your user or system config directory - you can find these at the top of `Help->Debug Messages`. Then just restart Geany. To recognise *.ts, you also have to edit `Tools->Config Files->filetype_extensions.conf`, see the *Files changed* tab here.

**Comment by jhpratt** [comment]
@ntrel I've long since moved on from Geany. I find VS Code far superior. It's been a couple years since I submitted this PR, so a good number of changes may be out of date as well. If you'd like this conflict resolved, you should be able to edit the PR. If not, feel free to use my changes at your discretion.

**Comment by ntrel** [comment]
implemented in #2187

**Pull request closed without merging**